test/membrane_http_adaptive_stream/integration_test/sink_bin_integration_test.exs

```diff
@@ -0,0 +1,192 @@
+    {%Source{
+       location:
+         "https://raw.githubusercontent.com/membraneframework/static/big_buck_bunny_samples/samples/big-buck-bunny/bun33s_480x270.h264",
+       hackney_opts: [follow_redirect: true]
+     }, :H264, "video_480x270"},
+    {%Source{
+       location:
+         "https://raw.githubusercontent.com/membraneframework/static/big_buck_bunny_samples/samples/big-buck-bunny/bun33s_540x360.h264",
+       hackney_opts: [follow_redirect: true]
+     }, :H264, "video_540x360"},
+    {%Source{
+       location:
+         "https://raw.githubusercontent.com/membraneframework/static/big_buck_bunny_samples/samples/big-buck-bunny/bun33s_720x480.h264",
+       hackney_opts: [follow_redirect: true]
+     }, :H264, "video_720x480"}
+  ]
+  @audio_multiple_video_tracks_ref_path "./test/membrane_http_adaptive_stream/integration_test/reference_playlists/audio_multiple_video_tracks/"
+  @audio_multiple_video_tracks_test_path "./test/membrane_http_adaptive_stream/integration_test/test_playlists/audio_multiple_video_tracks/"
+
+  defmodule TestPipeline do
+    use Membrane.Pipeline
+    alias Membrane.HTTPAdaptiveStream.{SinkBin, HLS}
+    alias Membrane.HTTPAdaptiveStream.Storages.FileStorage
+    alias Membrane.H264.FFmpeg.Parser, as: H264_Parser
+    alias Membrane.AAC.Parser, as: AAC_Parser
+
+    @impl true
+    def handle_init(options) do
+      sink_bin = %SinkBin{
+        muxer_segment_duration: 2 |> Membrane.Time.seconds(),
+        manifest_module: HLS,
+        target_window_duration: 30 |> Membrane.Time.seconds(),
+        target_segment_duration: 2 |> Membrane.Time.seconds(),
+        persist?: false,
+        storage: %FileStorage{
+          directory: options[:output_dir]
+        }
+      }
+
+      elements =
+        options[:sources]
+        |> Enum.map(fn {source, encoding, track_name} ->
+          [
+            {{:source, track_name}, source},
+            {{:parser, encoding, track_name}, get_parser_for_encoding(encoding)}
+          ]
+        end)
+
+      links =
+        elements
+        |> Enum.map(fn [
+                         {{:source, name} = source_id, _},
+                         {{:parser, encoding, name} = parser_id, _}
+                       ] ->
+          link(source_id)
+          |> to(parser_id)
+          |> via_in(Pad.ref(:input, name), options: [encoding: encoding, track_name: name])
+          |> to(:sink_bin)
+        end)
+
+      children = elements |> List.flatten() |> Enum.into(%{:sink_bin => sink_bin})
+
+      {{:ok, spec: %ParentSpec{children: children, links: links}}, %{}}
+    end
+
+    defp get_parser_for_encoding(encoding) do
+      case encoding do
+        :H264 ->
+          %H264_Parser{
+            framerate: {30, 1},
+            alignment: :au,
+            attach_nalus?: true
+          }
+
+        :AAC ->
+          %AAC_Parser{
+            out_encapsulation: :none
+          }
+      end
+    end
+  end
+
+  test "single video track" do
```

Comment:
test names should be more descriptive, if the test fails we really don't know what it was supposed to do, `describe` block is your friend.

Reply:
I have added detailed comments and explanations wherever I thought it was necessary. Please take a look if comments are readable and clear enough.